src/vdbfusion_ros/VDBVolume_ros.cpp

```diff
@@ -11,30 +11,42 @@
 #include <tf2_sensor_msgs/tf2_sensor_msgs.h>
 
 #include <Eigen/Core>
+#include <happly.h>
 #include <iostream>
 #include <vector>
+#include <array>
 
-#include "igl/write_triangle_mesh.h"
 #include "openvdb/openvdb.h"
 
 namespace {
-std::vector<Eigen::Vector3d> pcl2SensorMsgToEigen(const sensor_msgs::PointCloud2& pcl2) {
-    std::vector<Eigen::Vector3d> points;
-    points.reserve(pcl2.width);
-    sensor_msgs::PointCloud pcl;
-    sensor_msgs::convertPointCloud2ToPointCloud(pcl2, pcl);
+
+template <typename PCLPoint>
+inline bool isPointFinite(const PCLPoint& point) {
+  return std::isfinite(point.x) && std::isfinite(point.y) &&
+         std::isfinite(point.z);
+}
+
+void pclRgbToVec3i(pcl::PointCloud<pcl::PointXYZRGB>& pcl, std::vector<openvdb::Vec3i>& colors,
+    std::vector<Eigen::Vector3d>& points) {
+    colors.reserve(pcl.size());
+    points.reserve(pcl.size());
+    ROS_INFO("** Total of %ld RGBD points.", pcl.size());
```

Comment:
Since the code works pretty well, you can get rid of the `ROS_INFO()` you added for debugging.